### PR TITLE
docs: Add Operator docs under Kubernetes section

### PR DESCRIPTION
**Pull request by owenrumney**
## Description
Migrating the Trivy Operator docs into the hierarchy

Slightly rearranged the Operator docs to fit in more cleanly and hopefully a consistently

image

Signed-off-by: Owen Rumney <owen.rumney@aquasec.com>

### CODE REVIEW

File: docs/docs/kubernetes/operator/design/design_compliance_report.md [hunk 1]
```diff
@@ -0,0 +1,770 @@
+# Support Compliance Reports
```
Comment (itaysk) [inline docs/docs/kubernetes/operator/design/design_compliance_report.md:1]: we left compliance out of this release, so remove this

File: docs/docs/kubernetes/operator/configuration-auditing/index.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+continuously assess build artifacts, workloads, and infrastructure against configuration hardening standards to
+remediate any violations.
+
+trivy-operator configuration audit capabilities are purpose-built for Kubernetes environments. In particular, trivy
+Operator continuously checks images, workloads, and Kubernetes infrastructure components against common configurations
+security standards and generates detailed assessment reports, which are then stored in the default Kubernetes database.
+
+Kubernetes applications and other core configuration objects, such as Ingress, NetworkPolicy, ResourceQuota, RBAC
+resources, are evaluated against [Built-in Policies]. Beyond that, cluster nodes are constantly assessed against the CIS
+Kubernetes Benchmarks with the kube-bench [Infrastructure Scanner]. The results of all these scans are stored as
```
Comment (itaysk) [inline docs/docs/kubernetes/operator/configuration-auditing/index.md:15]: remove reference of CIS benchmarks

File: docs/docs/kubernetes/operator/configuration-auditing/index.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+[ConfigAuditReport], [ClusterConfigAuditReport], and [CISKubeBenchReport] resources, which could be further aggregated
+into a [ClusterComplianceReport] such as [NSA, CISA Kubernetes Hardening Guidance].
```
Comment (itaysk) [inline docs/docs/kubernetes/operator/configuration-auditing/index.md:17]: remove reference of compliance

File: docs/docs/kubernetes/operator/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+    Use `vuln` and `configaudit` as short names for `vulnerabilityreports` and `configauditreports` resources.
+
+!!! Note
+    You can define the validity period for VulnerabilityReports by setting the duration as the value of the
+    `OPERATOR_VULNERABILITY_SCANNER_REPORT_TTL` environment variable. For example, setting the value to `24h`
+    would delete reports after 24 hours. When a VulnerabilityReport gets deleted Trivy-Operator will automatically
+    rescan the underlying workload. Assuming that the vulnerability scanner has updated its vulnerability database,
+    new VulnerabilityReports will contain the latest vulnerabilities.
+
+## Infrastructure Scanning
```
Comment (itaysk) [inline docs/docs/kubernetes/operator/getting-started.md:189]: remove

File: docs/docs/kubernetes/operator/index.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+If hashes are not equal then affected ConfigAuditReport objects are deleted, which in turn triggers rescan - this time
+with new plugin's configuration.
+
+<figure>
+  <img src="./images/operator/trivy-operator-config.png" />
+  <figurecaption>Plugin configuration reconciler deletes ConfigAuditReports whenever the configuration changes.</figurecaption>
+</figure>
+
+!!! warning
+    Currently, the operator supports [vulnerabilityreports] and [configauditreports]. We plan to support [kubehunterreports] and [ciskubebenchreports] security resources soon. We also plan to implement rescan on configurable schedule, for
```
Comment (itaysk) [inline docs/docs/kubernetes/operator/index.md:28]: remove this

File: mkdocs.yml [hunk 6]
```diff
@@ -57,12 +57,34 @@ nav:
...
+                    - Helm: docs/kubernetes/operator/installation/helm.md
+                    - Operator Lifecycle Manager: docs/kubernetes/operator/installation/olm.md
+                    - Upgrade: docs/kubernetes/operator/installation/upgrade.md
+              - Getting Started: docs/kubernetes/operator/getting-started.md
+              - Configuration: docs/kubernetes/operator/configuration.md
+              - Settings: docs/kubernetes/operator/settings.md
+              - Vulnerability Scanning:
+                    - Overview: docs/kubernetes/operator/vulnerability-scanning/index.md
+                    - Trivy Scanner: docs/kubernetes/operator/vulnerability-scanning/trivy.md
+                    - Aqua Enterprise Scanner: docs/kubernetes/operator/vulnerability-scanning/aqua-enterprise.md
```
Comment (itaysk) [inline mkdocs.yml:75]: remove

**Review by itaysk — CHANGES_REQUESTED** [review]
caught a few leftovers from starboard. I think a simple scan for keywords will help us find all the irrelevant parts: kube-hunter, kube-bench, CIS, compliance, starboard, aqua

File: docs/docs/kubernetes/operator/design/design_compliance_report.md (hunk 1 above)
Comment (owenrumney) [reply docs/docs/kubernetes/operator/design/design_compliance_report.md]: Removed

File: docs/docs/kubernetes/operator/configuration-auditing/index.md (hunk 2 above)
Comment (owenrumney) [reply docs/docs/kubernetes/operator/configuration-auditing/index.md]: Removed

File: docs/docs/kubernetes/operator/configuration-auditing/index.md (hunk 3 above)
Comment (owenrumney) [reply docs/docs/kubernetes/operator/configuration-auditing/index.md]: Removed

File: docs/docs/kubernetes/operator/getting-started.md (hunk 4 above)
Comment (owenrumney) [reply docs/docs/kubernetes/operator/getting-started.md]: Removed

File: docs/docs/kubernetes/operator/index.md (hunk 5 above)
Comment (owenrumney) [reply docs/docs/kubernetes/operator/index.md]: Removed

File: mkdocs.yml (hunk 6 above)
Comment (owenrumney) [reply mkdocs.yml]: Removed

**Comment by owenrumney** [comment]
I don't know if Further Reading should be removed as well? It references online resources about Starboard

**Comment by itaysk** [comment]
sure, we can remove it.

Also, could you please add some note in the operator index.md about relationship to starboard operator? something like:

> Trivy Operator is based on existing Aqua OSS project - [Starboard](https://github.com/aquasecurity/starboard), and shares much of the design, principles and code with it. Existing content that relates to Starboard Operator might also be relevant for Trivy Operator. To learn more about the transition from Starboard from Trivy, see the [announcement discussion](https://github.com/aquasecurity/starboard/discussions/1173)

**Comment by owenrumney** [comment]
I think at this point where we're editing text someone who us worked on the project does it.

It's like me asking someone to update tfsec docs who has no knowledge of the relationship

**Comment by itaysk** [comment]
fair enough, I'll give it another round and this time just commit into your branch

**Comment by itaysk** [comment]
gave it another run, tried to simplify things. I would try to do it again later an probably converge some pages.
also, I think Images dir still has some work to do.

**Comment by chen-keinan** [comment]
we can remove this image : docs/docs/kubernetes/operator/images/node01-report.png
This report was produced from Starboard CLI which no longer exist

File: mkdocs.yml [hunk 7]
```diff
@@ -145,6 +163,9 @@ extra:
   version:
     method: mike
     provider: mike
+  var:
+    prev_git_tag: "v0.0.0"
+    chart_version: 0.0.1
```
Comment (chen-keinan) [inline mkdocs.yml:168]: This probably should be Trivy chart version , or I assume this section can be removed if it wasn't exist before

**Comment by itaysk** [comment]
@chen-keinan
https://github.com/aquasecurity/trivy/pull/2111/commits/92af7a07639a1e80e25a99aa33d8eb6597dcd216 - merge config and settings, I think it wasn't clear why we two very similar pages.
https://github.com/aquasecurity/trivy/pull/2111/commits/3053a5c746e3fd91ef4387fcdfc18bdcbc6d6082 - merge installation options, not sure this one is better than before LMK WDYT, we can also revert this one

**Comment by chen-keinan** [comment]
> 
- The setting and config merge looks OK, for now, probably we will have to do an additional review to validate content and order.
- The Install option merge is a little bit confusing, not fully clear when each install ends and if the other part is relevant to me, for example, if I choose helm or else

**Comment by itaysk** [comment]
ok, reverted it

File: docs/docs/kubernetes/operator/installation/helm.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+!!! danger
+    Deleting custom resource definitions will also delete all security reports generated by the operator.
+
+    ```
+    kubectl delete crd vulnerabilityreports.aquasecurity.github.io
+    kubectl delete crd clustervulnerabilityreports.aquasecurity.github.io
+    kubectl delete crd configauditreports.aquasecurity.github.io
+    kubectl delete crd ciskubebenchreports.aquasecurity.github.io
+    kubectl delete crd kubehunterreports.aquasecurity.github.io
```
Comment (chen-keinan) [inline docs/docs/kubernetes/operator/installation/helm.md:84]: remove the kube hunter line : `kubectl delete crd kubehunterreports.aquasecurity.github.io`

Comment (owenrumney) [reply docs/docs/kubernetes/operator/installation/helm.md:84]: should the one above it also be removed?

**Review by itaysk — APPROVED** [review]

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**